libs/net/dal/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -118,4 +119,23 @@ public static IServiceCollection AddStorageConfig(this IServiceCollection servic
 
         return services;
     }
+
+    /// <summary>
+    /// Add the S3 configuration to the service collection.
+    /// </summary>
+    /// <param name="services"></param>
+    /// <param name="config"></param>
+    /// <returns></returns>
+    public static IServiceCollection AddS3Config(this IServiceCollection services, IConfiguration config)
+    {
+        services.Configure<S3Options>(config.GetSection("S3"));
+        services.AddSingleton(sp => sp.GetRequiredService<IOptions<S3Options>>().Value);
```

Comment:
You don't need a singleton of the configuration values.  You may only need a singleton of the helper class.